lib/Elastica/Util.php

```diff
@@ -11,6 +11,73 @@
  */
 class Util
 {
+    /** @var array */
+    protected static $dateMathSymbols = ['<', '>', '/', '{', '}', '|', '+', ':', ','];
+
+    /** @var array */
+    protected static $escapedDateMathSymbols = ['%3C', '%3E', '%2F', '%7B', '%7D', '%7C', '%2B', '%3A', '%2C'];
+
+    /**
+     * Checks if date math is already escaped within request URI.
+     *
+     * @param string $requestUri
+     *
+     * @return bool
+     */
+    public static function isDateMathEscaped($requestUri)
+    {
+        // In practice, the only symbol that really needs to be escaped in URI is '/' => '%2F'
+        return false !== strpos($requestUri, '%2F');
+    }
+
+    /**
+     * Escapes date math symbols within request URI.
+     * @see https://www.elastic.co/guide/en/elasticsearch/reference/5.x/date-math-index-names.html
+     *
+     * @param string $requestUri
+     *
+     * @return string
+     */
+    public static function escapeDateMath($requestUri)
+    {
+        // No need to escape if a second slash is not present
+        if (empty($requestUri) || false === strpos($requestUri, '/', 1)) {
```

Comment:
I wanted to add a quick check if we even need to escape the path. This covers `/_bulk`, `/_analyze` and such cases - quickly exits because there is no escaping to be done.